### PR TITLE
VIH-9083 - Add Template to Publish NuGet Packages

### PR DESCRIPTION
**Pull request by v1ferrarij**
### JIRA link (if applicable) ###

[VIH-9083](https://tools.hmcts.net/jira/browse/VIH-9083)

### Change description ###

add template to add NuGet packages to azure artifacts.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: templates/dotnet/publish-nuget-packages.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+parameters:
+  - name: nugetProjectPath
+    type: string
+
+  - name: vstsFeed
+    type: string
+    default: 'Video Hearings/vh-packages'
```
Comment (chris-pateman) [inline templates/dotnet/publish-nuget-packages.yml:7]: Can we remove defaults referring to project specific things please

File: templates/dotnet/publish-nuget-packages.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  inputs:
+    command: restore
+    projects: ${{ parameters.nugetProjectPath }}/*.csproj
+    vstsFeed: ${{ parameters.vstsFeed }}
+    verbosityLevel: Minimal
+
+- task: DotNetCoreCLI@2
+  displayName: Create Build NuGet Project
+  inputs:
+    arguments: '--configuration release /p:Version="$(MajorMinorPatch)" /p:PackageOutputPath="$(Build.ArtifactStagingDirectory)" /p:VersionSuffix="-alpha"'
```
Comment (chris-pateman) [inline templates/dotnet/publish-nuget-packages.yml:29]: I think the suffix is what we will need to use later for the other project, so might be worth making this a parameter

**Review by chris-pateman — CHANGES_REQUESTED** [review]

File: templates/dotnet/publish-nuget-packages.yml (hunk 2 above)
Comment (v1ferrarij) [reply templates/dotnet/publish-nuget-packages.yml]: I don't think the suffix has any effect on what is published to Azure DevOps, I cannot see the reference Alpha anywhere with all the new packages I have deployed yesterday, it looks like it is just Version= which is used.

I have made the Version an optional parameter for now, defaulting to use GitVersion if one is not specified.

File: templates/dotnet/publish-nuget-packages.yml (hunk 1 above)
Comment (v1ferrarij) [reply templates/dotnet/publish-nuget-packages.yml]: Done.

**Review by chris-pateman — APPROVED** [review]
LGTM

**Pull request merged**